### PR TITLE
[14.0][FIX] account_operating_unit: hook function and check inter ou on company

### PR DESCRIPTION
**Pull request by Saran440**
move on function add moveline on payment to new function
and check inter ou on company, if not it can overwrite line without change stage.

### CODE REVIEW

File: account_operating_unit/wizards/account_payment_register.py [hunk 1]
```diff
@@ -7,32 +7,37 @@
 class AccountPaymentRegister(models.TransientModel):
     _inherit = "account.payment.register"
 
+    def _update_moveline_operating_unit(self, payments, payment, reconciled_moves):
+        to_reconcile = self.env["account.move.line"]
+        destination_account = payments.destination_account_id
```
Comment (AaronHForgeFlow) [inline account_operating_unit/wizards/account_payment_register.py:12]: Is it possible to get a singleton error here?

**Review by AaronHForgeFlow — COMMENTED** [review]
The hook is fine to me. Just a minor comment.

**Comment by Saran440** [comment]
@AaronHForgeFlow Thank you for your comment.

~~Do you know why this module change state payment before overwrite operating_unit?~~ -> For multi invoices

**Comment by AaronHForgeFlow** [comment]
@Saran440  I was referring to the line `destination_account = payments.destination_account_id` did you test it works when several payments are available?

**Comment by Saran440** [comment]
@AaronHForgeFlow

You can choose 1 journal when you register your payment. So, it's work with several payments.

BUT I tested 2 vendor bills (difference OU) register 1 payment  -->  **it's error.**

**Sample**
Vendor Bills A  (OU: B2B)
Vendor Bills B (OU: B2C)
Journal Bank (OU: Main Operating Unit)
and register payment with group payments -> it's error


-----------------------------------------------------------

Should we fix it on other PR or here?

**Pull request closed without merging**